dataline-workers/src/test/java/io/dataline/workers/DefaultSyncWorkerTest.java

```diff
@@ -0,0 +1,171 @@
+  private static final long LAST_SYNC_TIME = 1598565106;
+  private static final UUID WORKSPACE_ID = UUID.randomUUID();
+  private static final UUID SOURCE_SPECIFICATION_ID = UUID.randomUUID();
+  private static final UUID SOURCE_IMPLEMENTATION_ID = UUID.randomUUID();
+  private static final UUID DESTINATION_SPECIFICATION_ID = UUID.randomUUID();
+  private static final UUID DESTINATION_IMPLEMENTATION_ID = UUID.randomUUID();
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void test() throws Exception {
+    final ObjectMapper objectMapper = new ObjectMapper();
+
+    final ObjectNode sourceConnection = objectMapper.createObjectNode();
+    sourceConnection.put("apiKey", "123");
+    sourceConnection.put("region", "us-east");
+
+    final ObjectNode destinationConnection = objectMapper.createObjectNode();
+    destinationConnection.put("username", "dataline");
+    destinationConnection.put("token", "anau81b");
+
+    final SourceConnectionImplementation sourceConnectionConfig =
+        new SourceConnectionImplementation();
+    sourceConnectionConfig.setConfiguration(sourceConnection);
+    sourceConnectionConfig.setWorkspaceId(WORKSPACE_ID);
+    sourceConnectionConfig.setSourceSpecificationId(SOURCE_SPECIFICATION_ID);
+    sourceConnectionConfig.setSourceImplementationId(SOURCE_IMPLEMENTATION_ID);
+    sourceConnectionConfig.setTombstone(false);
+
+    final DestinationConnectionImplementation destinationConnectionConfig =
+        new DestinationConnectionImplementation();
+    destinationConnectionConfig.setConfiguration(destinationConnection);
+    destinationConnectionConfig.setWorkspaceId(WORKSPACE_ID);
+    destinationConnectionConfig.setDestinationSpecificationId(DESTINATION_SPECIFICATION_ID);
+    destinationConnectionConfig.setDestinationImplementationId(DESTINATION_IMPLEMENTATION_ID);
+
+    final Column column = new Column();
+    column.setName(COLUMN_NAME);
+    column.setDataType(DataType.STRING);
+    column.setSelected(true);
+
+    final Table table = new Table();
+    table.setName(TABLE_NAME);
+    table.setSelected(true);
+    table.setColumns(Collections.singletonList(column));
+
+    final Schema schema = new Schema();
+    schema.setTables(Collections.singletonList(table));
+
+    final UUID connectionId = UUID.randomUUID();
+    StandardSync standardSync = new StandardSync();
+    standardSync.setConnectionId(connectionId);
+    standardSync.setDestinationImplementationId(DESTINATION_IMPLEMENTATION_ID);
+    standardSync.setSourceImplementationId(SOURCE_IMPLEMENTATION_ID);
+    standardSync.setStatus(StandardSync.Status.ACTIVE);
+    standardSync.setSyncMode(StandardSync.SyncMode.APPEND);
+    standardSync.setName("favorite_color_pipe");
+    standardSync.setSchema(schema);
+
+    final ObjectNode stateValue = objectMapper.createObjectNode();
+    stateValue.put("lastSync", LAST_SYNC_TIME);
+
+    State state = new State();
+    state.setConnectionId(connectionId);
+    state.setState(stateValue);
```

Comment:
I know it doesn't belong to this PR but fields that contain serialized data should remained in their serialized form (`string`) and the responsibility of deserizliazing belongs to the consumer of that blob (for state I assume it should only be in the integration code).

Putting `ObjectNode` is going to lead to a lot of confusion in the future since it is not typed and the name doesn't describe the type.

Reply:
yeah. 100% agree. ran into this already in the unit testing. will be careful about this as i go back through and test.

Reply:
and will fix here.